data_structures/linked_list.cpp

```diff
@@ -1,58 +1,152 @@
+/**
+ * @file linked_list.cpp
+ * @brief Implementation of singly linked list algorithm.
+ *
+ * @details
+ * The linked list is a form of container used for holding a sequence of 
+ * values, which can be added, removed and displayed.
+ * ### Algorithm
+ * Values can be added by iterating to the end of a list(by following 
+ * the pointers) starting from the first node. Whichever node points to null
+ * is considered the last node and is pointed to the new value.
+ *
+ * Values can be removed by also iterating through the list. When the node containing 
+ * the value is found, the node pointing to the current node is made to point to the 
+ * node that the current node is pointing to, and then returning the current node to
+ * heap store.
+ */
 #include <iostream>
+#include <sstream>
 
-struct node {
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures {
+/**
+ * stores value of a node and address of the next node
+ */
+struct node
+{
     int val;
     node *next;
 };
+	
+/**
+ * This function checks if the string passed consists 
+ * of only digits.
+ * @param s To be checked if s contains only integers
+ * @returns true if there are only only digits present in the string
+ * @returns false if any other character is found
+ */
+bool isDigit(const std::string& s)
+{
+	// function statements here
+	for(char i:s)
+    {
+    	if(!isdigit(i))
+    	{
+            return false;
+		}
+	}
+	return true;
+}
+	
+/**
+ * This function changes the value_type of its argument
+ * from a string to an integer
+ * @param s argument whose value_type we want to change
+ * @return number value with value_type as integer
+ */
+int toInt(const std::string& s)
+{
+	// function statements here
+	std::stringstream geek(s);
+	int number=0;
+	geek>>number;
+	return number;
+} 
 
-node *start;
+typedef node* Iter;
 
-void insert(int x) {
-    node *t = start;
-    node *n = new node;
-    n->val = x;
-    n->next = NULL;
-    if (start != NULL) {
-        while (t->next != NULL) {
+/**
+ * This function creates a new node using the new value
+ * passed as its argument and inserts it to the end of the 
+ * list.
+ * @param x value we want to insert to the list
+ * @param start first node of the list we want to insert to
+ */
+void insert(int x, Iter* start) {
+	// function statements here
+    node* t = *start;
+    node* n = new node;// creates new node
```

Comment:
The memory is not freed, and may lead to memory leaks.
After use, please free up the allocated memory using `delete`.

Reply:
line 275-280 frees all the allocated memor